core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java

```diff
@@ -27,31 +32,34 @@ public PlayerEntity(World world, PlayerDef def) {
         previousState = PlayerState.STANDING;
         verticalDirection = LookingDirection.RIGHT;
         horizontalDirection = null;
-        this.def = def;
+        inventory = def.inventory;
+        dragModifier = def.dragModifier;
+        maxSpeedModifier = def.maxSpeedModifier;
+        textureVariant = def.textureVariant;
     }
 
     public float getDragModifier() {
-        return def.dragModifier;
+        return dragModifier;
     }
 
-    public void setDragModifier(float dragModifier) {
-        def.dragModifier = dragModifier;
+    public void setDragModifier(float newDragModifier) {
+        dragModifier = newDragModifier;
     }
 
     public float getMaxSpeedModifier() {
-        return def.maxSpeedModifier * def.inventory.getMovementSpeedMultiplier();
+        return maxSpeedModifier * inventory.getMovementSpeedMultiplier();
     }
 
-    public void setMaxSpeedModifier(float maxSpeedModifier) {
-        def.maxSpeedModifier = maxSpeedModifier;
+    public void setMaxSpeedModifier(float newMaxSpeedModifier) {
+        maxSpeedModifier = newMaxSpeedModifier;
```

Comment:
Rename to match other getters and use this.maxSpeedModifer to assign

Reply:
Fixed